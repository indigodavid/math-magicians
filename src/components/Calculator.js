import { useState } from 'react';
import CalculatorButtons from './CalculatorButtons';
import CalculatorDisplay from './CalculatorDisplay';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [obj, setObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleChange = (input) => {
    setObj(calculate(obj, input));
    return obj;
  };

  const getDisplayInfo = () => obj;

  return (
    <div id="calculator">
      <CalculatorDisplay onDisplay={getDisplayInfo} onKeyEvent={handleChange} />
      <CalculatorButtons onClickButtons={handleChange} />
    </div>
  );
};

export default Calculator;
