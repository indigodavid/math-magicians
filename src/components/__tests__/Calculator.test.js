import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../Calculator';

describe('Tests for Calculator Module', () => {
  const title = "Let's do some math!"
  render(<Calculator />)
  test('The calculator component is displayed', () => {
    expect(screen.getByText(title)).toBeInTheDocument();
  })

  test('Display updates when buttons are pressed', () => {
    const { container } = render(<Calculator />);
    const button0 = container.querySelector('#button-0');
    const button1 = container.querySelector('#button-1');
    const display = container.querySelector('#input-display');
    fireEvent.click(button1);
    expect(display.value).toBe('1');
    fireEvent.click(button0);
    expect(display.value).toBe('10');
  })

  test('Test number of buttons', async () => {
    render(<Calculator />);
    const buttonList = await screen.findAllByRole('button');
    expect(buttonList).toHaveLength(19);
  });

})