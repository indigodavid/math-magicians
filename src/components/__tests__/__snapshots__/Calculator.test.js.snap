// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Calculator Module The calculator component is displayed 1`] = `
<div
  id="wrapper"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    id="calculator"
  >
    <div
      class="display"
    >
      <input
        id="input-display"
        name="input-display"
        type="text"
        value="0"
      />
    </div>
    <div
      id="calculator-buttons"
    >
      <button
        class="calculator-button"
        id="button-ac"
        type="button"
      >
        AC
      </button>
      <button
        class="calculator-button"
        id="button-sign"
        type="button"
      >
        +/-
      </button>
      <button
        class="calculator-button"
        id="button-percent"
        type="button"
      >
        %
      </button>
      <button
        class="calculator-button button-orange"
        id="button-divide"
        type="button"
      >
        รท
      </button>
      <button
        class="calculator-button"
        id="button-7"
        type="button"
      >
        7
      </button>
      <button
        class="calculator-button"
        id="button-8"
        type="button"
      >
        8
      </button>
      <button
        class="calculator-button"
        id="button-9"
        type="button"
      >
        9
      </button>
      <button
        class="calculator-button button-orange"
        id="button-multiply"
        type="button"
      >
        x
      </button>
      <button
        class="calculator-button"
        id="button-4"
        type="button"
      >
        4
      </button>
      <button
        class="calculator-button"
        id="button-5"
        type="button"
      >
        5
      </button>
      <button
        class="calculator-button"
        id="button-6"
        type="button"
      >
        6
      </button>
      <button
        class="calculator-button button-orange"
        id="button-substract"
        type="button"
      >
        -
      </button>
      <button
        class="calculator-button"
        id="button-1"
        type="button"
      >
        1
      </button>
      <button
        class="calculator-button"
        id="button-2"
        type="button"
      >
        2
      </button>
      <button
        class="calculator-button"
        id="button-3"
        type="button"
      >
        3
      </button>
      <button
        class="calculator-button button-orange"
        id="button-add"
        type="button"
      >
        +
      </button>
      <button
        class="calculator-button button-zero"
        id="button-0"
        type="button"
      >
        0
      </button>
      <button
        class="calculator-button"
        id="button-."
        type="button"
      >
        .
      </button>
      <button
        class="calculator-button button-orange"
        id="button-equal"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
